### PR TITLE
[flutter drive] Do not start dds if --no-dds

### PR DESCRIPTION
**Pull request by dnfield**
Lessens severity of https://github.com/flutter/flutter/issues/81707

Make sure we respect `--no-dds` on `flutter drive`. Without this, we unconditionally launch DDS.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/drive/drive_service.dart [hunk 1]
```diff
@@ -204,19 +204,21 @@ class FlutterDriverService extends DriverService {
...
+          ipv6,
+          debuggingOptions.disableServiceAuthCodes,
+          logger: _logger,
+        );
+        _vmServiceUri = device.dds.uri.toString();
+      } on dds.DartDevelopmentServiceException {
+        // If there's another flutter_tools instance still connected to the target
+        // application, DDS will already be running remotely and this call will fail.
+        // This can be ignored to continue to use the existing remote DDS instance.
+      }
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/drive/drive_service.dart:221]: maybe we should printTrace something here? up to you

**Review by Hixie — APPROVED** [review]

File: packages/flutter_tools/lib/src/drive/drive_service.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/lib/src/drive/drive_service.dart:221]: This call already prints some traces when it tries to launch, as well as when it fails (and rethrows).

**Comment by dnfield** [comment]
@jonahwilliams - apparently we already had a test that thought it was asserting this but wasn't. And for reasons I don't undertand, this change made that test's assertions around VM service URI different. Is it valid to update it like this?

**Review by jonahwilliams — APPROVED** [review]
As long as it works IRL the test update should be fine

**Pull request merged**